### PR TITLE
Tests for support of different cryptographic algorhitms

### PR DESCRIPTION
**Pull request by tym2k1**

### CODE REVIEW

**Comment by TomaszAIR** [comment]
@tym2k1 should this still be in draft? Please also rebase onto `uefi-sb-release` and change target branch for this PR.

File: dasharo-security/secure-boot.robot [hunk 1]
```diff
@@ -233,6 +233,108 @@ SBO008.001 Attempt to enroll the key in the incorrect format (firmware)
     Select File In File Explorer    DB.txt
     Read From Terminal Until    ERROR: Unsupported file type!
 
+SBO009.001 Check support for rsa2k signed certificates
+    Download ISO And Mount As USB    ${RSA2_K_TEST_NAME}    ${RSA2_K_TEST_URL}
```
Comment (TomaszAIR) [inline dasharo-security/secure-boot.robot:237]: @tym2k1 right now I got this for SBO009.001

```
λ robot -L TRACE -v rte_ip:127.0.0.1 -v snipeit:no -v config:qemu -t "SBO009.001*" dasharo-security/secure-boot.robot
==============================================================================
Secure-Boot                                                                   
==============================================================================
SBO009.001 Check support for rsa2k signed certificates                | FAIL |
Keyword 'usb-hid-msc-lib.Download ISO And Mount As USB' expected 3 arguments, got 2.
------------------------------------------------------------------------------
```

This keyword is missing SHA as an argument. Also, can we use similar syntax to this one https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/dasharo-security/secure-boot.robot#L190 - can we download this image to `DL_CACHE_DIR` ?

File: dasharo-security/secure-boot.robot [hunk 2]
```diff
@@ -233,6 +233,108 @@ SBO008.001 Attempt to enroll the key in the incorrect format (firmware)
     Select File In File Explorer    DB.txt
     Read From Terminal Until    ERROR: Unsupported file type!
 
+SBO009.001 Check support for rsa2k signed certificates
```
Comment (TomaszAIR) [inline dasharo-security/secure-boot.robot:236]: @tym2k1 please add short test description for every test case similar to https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/dasharo-security/secure-boot.robot#L185; explain there how we are creating the given keys

File: dasharo-security/secure-boot.robot [hunk 3]
```diff
@@ -253,4 +355,28 @@ Prepare Test Files
         ...    ${BAD_FORMAT_NAME}
         ...    ${BAD_FORMAT_URL}
         ...    ${BAD_FORMAT_SHA256}
+        Download To Host Cache
```
Comment (TomaszAIR) [inline dasharo-security/secure-boot.robot:358]: @tym2k1 I do not think that `Prepare Test Files` keyword is used anywhere, those changes are not needed. Now, we are getting files using `Download ISO And Mount As USB`

**Comment by tym2k1** [comment]
My bad. These changes were developed for the now merged `qemu-rebase-pt2` and I forgot to rerun and adress changes to syntax after rebasing.

File: dasharo-security/secure-boot.robot (hunk 1 above)
Comment (tym2k1) [reply dasharo-security/secure-boot.robot]: Addressed and squashed. The `robotidy` hook of pre-commit forced multiline syntax for `ecdsa256`, `ecdsa384` and `ecdsa521`

File: dasharo-security/secure-boot.robot (hunk 2 above)
Comment (tym2k1) [reply dasharo-security/secure-boot.robot:237]: Addressed and squashed

File: dasharo-security/secure-boot.robot (hunk 3 above)
Comment (tym2k1) [reply dasharo-security/secure-boot.robot]: Addressed and squashed

**Pull request merged**